ledger/ledgercore/persistedacctdata.go

```diff
@@ -43,52 +47,91 @@ type AssetsHoldingGroup struct {
 	// A foreign key to the accountext table to the appropriate AssetsHoldingGroupData entry
 	// AssetGroupKey is 0 for newly created entries and filled after persisting to DB
 	AssetGroupKey int64 `codec:"k"`
+}
 
+// AssetsHoldingGroup is a metadata for asset group data (AssetsHoldingGroupData)
+// that is stored separately
+type AssetsHoldingGroup struct {
+	AssetGroupDesc
 	// groupData is an actual group data
 	groupData AssetsHoldingGroupData
+	// loaded indicates either groupData loaded or not
+	loaded bool
+}
 
+// AssetsParamsGroup is a metadata for asset group data (AssetsParamsGroupData)
+// that is stored separately
+type AssetsParamsGroup struct {
+	AssetGroupDesc
+	// groupData is an actual group data
+	groupData AssetsParamsGroupData
 	// loaded indicates either groupData loaded or not
 	loaded bool
 }
 
-// AssetsHoldingGroupData is an actual asset holding data
-type AssetsHoldingGroupData struct {
+// AssetsCommonGroupData is common data type for Holding and Params storing AssetOffsets data
+type AssetsCommonGroupData struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
 	// offset relative to MinAssetIndex and differential afterward
-	// assetId1 = AmountsAssetIndicesOffsets[0] + MinAssetIndex and assetIdx1 == MinAssetIndex
-	// assetId2 = AmountsAssetIndicesOffsets[1] + assetIdx1
-	// assetId3 = AmountsAssetIndicesOffsets[2] + assetIdx2
+	// assetId1 = AssetOffsets[0] + MinAssetIndex and assetIdx1 == MinAssetIndex
+	// assetId2 = AssetOffsets[1] + assetIdx1
+	// assetId3 = AssetOffsets[2] + assetIdx2
 	AssetOffsets []basics.AssetIndex `codec:"ao,allocbound=MaxHoldingGroupSize"`
+}
 
-	// Holding amount
-	// same number of elements as in AmountsAssetIndicesOffsets
-	Amounts []uint64 `codec:"a,allocbound=MaxHoldingGroupSize"`
+// AssetsParamsGroupData is an actual asset param data
+type AssetsParamsGroupData struct {
+	AssetsCommonGroupData
+
+	// same number of elements as in AssetOffsets
+	Totals         []uint64         `codec:"t,allocbound=MaxParamsGroupSize"`
+	Decimals       []uint32         `codec:"d,allocbound=MaxParamsGroupSize"`
+	DefaultFrozens []bool           `codec:"f,allocbound=MaxParamsGroupSize"`
+	UnitNames      []string         `codec:"u,allocbound=MaxParamsGroupSize"`
+	AssetNames     []string         `codec:"n,allocbound=MaxParamsGroupSize"`
+	URLs           []string         `codec:"l,allocbound=MaxParamsGroupSize"`
+	MetadataHash   [][32]byte       `codec:"h,allocbound=MaxParamsGroupSize"`
+	Managers       []basics.Address `codec:"m,allocbound=MaxParamsGroupSize"`
+	Reserves       []basics.Address `codec:"r,allocbound=MaxParamsGroupSize"`
+	Freezes        []basics.Address `codec:"z,allocbound=MaxParamsGroupSize"`
+	Clawbacks      []basics.Address `codec:"c,allocbound=MaxParamsGroupSize"`
```

Comment:
you might want to take a look on Nicholas's work - specifically the bitset implementation.